libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetectorTest.java

```diff
@@ -89,6 +89,26 @@ public void isUserOffRoute_AssertTrueWhenTooFarFromStep() throws Exception {
     assertTrue(isUserOffRoute);
   }
 
+  @Test
+  public void isUserOffRoute_StepPointSize() throws Exception {
```

Comment:
Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert
If the `A`rrange part gets too long we should create factory or builder methods.

Reply:
Good point, I thought I had to delete the points, after the first off route check, it turns out, it's ok to do it before as well. Changed the order.

Reply:
NIT - You can also remove some "unnecessary" new lines so that the 3 parts of the test (`A`rrange, `A`ct and `A`ssert) are visually 👌 i.e.
```java
@Test
public void test() {
  RouteProgress routeProgress = buildDefaultTestRouteProgress();
  // ...
  
  boolean isUserOffRoute = offRouteDetector.isUserOffRoute(secondUpdate, routeProgress, options);
  
  assertFalse(isUserOffRoute);
}
```

Reply:
Sure that's not a problem.